tools/clang/test/HLSLFileCheck/hlsl/objects/ByteAddressBuffer/bab_mat_load_method.hlsl

```diff
@@ -0,0 +1,54 @@
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=float %s | FileCheck %s -check-prefix=CHK_SCALAR
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=float1 %s | FileCheck %s -check-prefix=CHK_VEC1
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=float1x1 %s | FileCheck %s -check-prefix=CHK_MAT1x1
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=bool1x2 %s | FileCheck %s -check-prefix=CHK_MAT1x2
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=int2x1 %s | FileCheck %s -check-prefix=CHK_MAT2x1
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=uint2x2 %s | FileCheck %s -check-prefix=CHK_MAT2x2
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=float16_t2x3 %s | FileCheck %s -check-prefix=CHK_MAT2x3
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=uint16_t3x2 %s | FileCheck %s -check-prefix=CHK_MAT3x2
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=float3x3 %s | FileCheck %s -check-prefix=CHK_MAT3x3
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=int3x4 %s | FileCheck %s -check-prefix=CHK_MAT3x4
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=bool4x3 %s | FileCheck %s -check-prefix=CHK_MAT4x3
+// RUN: %dxc -E main -T vs_6_2 -enable-16bit-types -DTY=uint4x4 %s | FileCheck %s -check-prefix=CHK_MAT4x4
```

Comment:
No 64-bit types tested here.  Need to test SM 6.3+ as well, since that introduces 64-bit RawBufferLoad/Store.  Also need to test min-precision case.